### ISSUE TITLE
Python 'pendulum' cannot be converted to a MySQL type

### ISSUE THREAD

**Issue by orzZXY** [apache/airflow#10795]
**Apache Airflow version**: 1.10.10

**Kubernetes version (if you are using kubernetes)** (use `kubectl version`):

**Environment**: python 3.7.6

- **Cloud provider or hardware configuration**: AWS m4.xlarge
- **OS** (e.g. from /etc/os-release): Amazon Linux
- **Kernel** (e.g. `uname -a`): Linux ****.amzn2.x86_64 #1 SMP *** UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
- **Install tools**: pip3
- **Others**:

**What happened**:

airflow backfill your_dag –s yyyy-mm-ddTHH:MM:00 –e yyyy-mm-ddTHH:MM:00` it would return the error `Python 'pendulum' cannot be converted to a MySQL type`, however, if you run the same command again, it would succeed. 

In other words, the error happens every first time you run the command.
Which means, though there is an error raised when you query/update the MySQL cluster, somehow it actually works

A solution that I figured out:
You need to modify /usr/local/lib/python3.7/site-packages/airflow/models/dagrun.py 
Replace exec_date = func.cast(self.execution_date, DateTime) TO
from airflow.utils.timezone import make_naive
exec_date = func.cast(make_naive(self.execution_date), DateTime)

**Comment by eladkal** [comment]
Hi @orzZXY can you open a PR to fix the bug?

**Comment by ashb** [comment]
Please give a stack trace for this full error.

**Comment by crawlersick** [comment]
this is the mysql-connector 's fault.
we can add the convert of pendulum to it's code just like datetime.
step1. find the /mysql/connector/conversion.py souce code in the system, maybe under the /usr/lib64/python3.x/site-packages , just depends on your python install.
step2. modify this python code conversion.py , just copy the code of method **_datetime_to_mysql,** to new a method named as **_pendulum_to_mysql**

**Comment by eladkal** [comment]
Closing as this seems not to be issue with Airflow.
We can revisit if there will be a reason to handle it from Airflow side.
